core/src/main/java/org/hibernate/ogm/util/impl/ArrayHelper.java

```diff
@@ -86,4 +87,20 @@ public static boolean contains(Object[] array, Object element) {
 
 		return result;
 	}
+
+	public static <T> T[] concat(List<T[]> arrays) {
```

Comment:
Javadoc?

BTW the ideal polishing in my opinion would be to not need such things :) Are you sure we can't avoid all this copying around?